lib/Models/CatalogItem.js

```diff
@@ -367,19 +347,35 @@ CatalogItem.defaultUpdaters.rectangle = function(catalogItem, json, propertyName
 };
 
 CatalogItem.defaultUpdaters.attribution = function(catalogItem, json, prototypeName) {
-    if(defined(json.attribution)){
-        if((typeof json.attribution === 'object' && json.attribution.text) && json.attribution.link ){
+    if (defined(json.attribution)) {
+        if ((typeof json.attribution === 'object' && json.attribution.text) && json.attribution.link) {
             catalogItem.attribution = new Credit(json.attribution.text, undefined, json.attribution.link);
-        }else if(typeof json.attribution === 'string'){
+        } else if (typeof json.attribution === 'string') {
             catalogItem.attribution = new Credit(json.attribution, undefined, undefined);
         }
     }
 };
+
+CatalogItem.defaultUpdaters.legendUrl = function(catalogItem, json, prototypeName) {
+    if (json.legendUrl) {
+        catalogItem.legendUrls = [new LegendUrl(json.legendUrl)];
+    }
+};
+
+CatalogItem.defaultUpdaters.legendUrls = function(catalogItem, json, prototypeName) {
+    if (defined(json.legendUrls)) {
+        catalogItem.legendUrls = json.legendUrls && json.legendUrls.map(function(rawLegend) {
+            return new LegendUrl(rawLegend.url, rawLegend._mimeType);
```

Comment:
Yeah - the logic for deciding whether a certain mimeType is actually an image lives in `LegendUrl.prototype.isImage`.